inline_test.go

```diff
@@ -570,13 +570,16 @@ func TestInlineLink(t *testing.T) {
 
 		// no closing ), autolinking detects the url next
 		"[disambiguation](http://en.wikipedia.org/wiki/Disambiguation_(disambiguation) is the",
-		"<p>[disambiguation](<a href=\"http://en.wikipedia.org/wiki/Disambiguation_(disambiguation\">http://en.wikipedia.org/wiki/Disambiguation_(disambiguation</a>) is the</p>\n",
+		"<p>[disambiguation](<a href=\"http://en.wikipedia.org/wiki/Disambiguation_(disambiguation)\">http://en.wikipedia.org/wiki/Disambiguation_(disambiguation)</a> is the</p>\n",
```

Comment:
Looks like the section you're referring to talks about the "fully qualified" links, the ones with full link syntax.

And it [seems](http://spec.commonmark.org/0.27/#autolinks) that CommonMark only has autolinks with a hint: they require angle brackets around the link, which avoids the balancing problem altogether.